core/src/net/cidr/v4.rs

```diff
@@ -53,20 +115,49 @@ impl Cidr for Ipv4Cidr {
         self.address
     }
 
+    #[inline]
+    fn contains(&self, address: Self::Addr) -> bool {
+        self.prefix == (self.mask & u32::from(address))
+    }
+
+    #[inline]
+    fn network(&self) -> Self::Addr {
+        Ipv4Addr::from(self.mask & u32::from(self.address))
+    }
+
+    #[inline]
+    fn broadcast(&self) -> Self::Addr {
+        Ipv4Addr::from((self.mask & u32::from(self.address)) | !self.mask)
+    }
+
     #[inline]
     fn length(&self) -> usize {
         self.length
```

Comment:
`length` is just `mask.leading_zeros()` right? we can save a few bytes by not storing it.